ksy_schema.json

```
@@ -270,6 +270,59 @@
         "^-.*$": true
       }
     },
+    "TypeRef": {
+      "anyOf": [
+        {
+          "description": "name of or path to existing type name\n\nif name is single path element, then type resolved in such order:\n- from subtypes of current type\n- current type (so, recursive types is possible)\n- parent type\n\none can reference a nested user-defined type by specifying a relative path to it from the current type, with a double colon as a path delimiter (e.g. `foo::bar::my_type`)",
+          "type": "string",
+          "pattern": "^([a-z][a-z0-9_]*::)*[a-z][a-z0-9_]*$"
```

This pattern doesn't allow type parameters. Something like this should work:

```suggestion
          "pattern": "^([a-z][a-z0-9_]*::)*[a-z][a-z0-9_]*(\\(.+\\))?$"
```